### PR TITLE
fix: Delete deprecated properties from UIScreen+UISafeArea.swift - WPB-8778

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8778" title="WPB-8778" target="_blank">TaskWPB-8778</a>  Clean up code base
  </td></table>
  <br />
 

---

#### Description:

This PR removes the entire file that contained deprecated extensions for `UIScreen`, which provided functionality for calculating safe area insets and notch detection. These functions have been marked for removal, and the usage of `safeAreaInsets` from `UIView` is now preferred.

#### Key Changes:

- Deleted the entire file, as it was no longer necessary.
- Deprecated methods removed include:
  - `safeArea`: Previously used for safe area insets; replaced by `UIView.safeAreaInsets`.
  - `hasBottomInset`: Used to detect bottom insets; removed without replacement.
  - `hasNotch`: Used to check for a device notch; no longer needed.
  - `isCompact`: Used to determine if the device was in compact mode; removed.

#### Testing:
- Verified that `UIView.safeAreaInsets` is being used across the codebase where necessary.
- Ensured no build or runtime errors occur after the removal of this file.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png:1]: It this change intentional

File: wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent_CameraAccessGranted.1.png
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent_CameraAccessGranted.1.png:1]: It this change intentional

**Review by samwyndham — COMMENTED** [review]
Hey @agisilaos, there are a lot of changes to snapshot tests where things have clearly changed position. Are these intentional?

**Comment by agisilaos** [comment]
Yep they're intentional. Now they're represented as they should in the snapshot tests. @samwyndham

File: wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testInitialStateLayoutSizeCompact_LibraryAccessGranted.1.png
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testInitialStateLayoutSizeCompact_LibraryAccessGranted.1.png:1]: The icon on the right is outside the safe area on my iPhone 13 device

**Review by samwyndham — CHANGES_REQUESTED** [review]
I had another pass over using my iPhone 13. I found one issue where an icon is outside the safe area. See comment.

**Review by samwyndham — APPROVED** [review]
Nice work!

Just some small feedback I was considering due to the changes in snapshot tests. 

I think one of the main benefits of any kind of automated testing is to provide safety when refactoring. This PR consists of refactoring + behavior change (small changes in UI appearance) + test improvements. Because of this we loose the safety provided by the tests. 

I think the PR process could have been smoother if it was dived into 2 PRs. The first purely refactoring where no snapshot tests should fail. This ensures that nothing has broken unexpectedly by the refactoring. A second PR could address any UI updates and improvements to tests.

**Comment by agisilaos** [comment]
> I think the PR process could have been smoother if it was dived into 2 PRs. The first purely refactoring where no snapshot tests should fail. This ensures that nothing has broken unexpectedly by the refactoring. A second PR could address any UI updates and improvements to tests.

That wouldn't be possible in this case. Because changes related to constraints impact snapshot tests. So there is no way I would make changes in the constraints and for snapshot tests to not fail. But I get your point, that would be the optimal if it was possible.

File: wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png
Comment (caldrian) [inline wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png:1]: Screenshot 2024-09-24 at 11 13 23

optional: On devices with home button the bottom space is probably a bit more than expected.
Did you check these screens on a device with home button?

Comment (caldrian) [reply wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png:1]: hm, on a device without home button I wonder if the camera preview should reach the bottom of the screen.
Screenshot 2024-09-24 at 11 17 45

File: wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController+Constraints.swift [hunk 1]
```diff
@@ -79,19 +77,21 @@ extension ContactsViewController {
...
-        let bottomEdgeConstraint = inviteOthersButton.bottomAnchor.constraint(equalTo: inviteOthersButton.superview!.bottomAnchor, constant: -(standardOffset / 2.0 + UIScreen.safeArea.bottom))
-        self.bottomEdgeConstraint = bottomEdgeConstraint
-
-        constraints += [
-            bottomEdgeConstraint,
-            inviteOthersButton.topAnchor.constraint(equalTo: inviteOthersButton.superview!.topAnchor, constant: standardOffset / CGFloat(2)),
-            inviteOthersButton.leadingAnchor.constraint(equalTo: inviteOthersButton.superview!.leadingAnchor, constant: standardOffset),
-            inviteOthersButton.trailingAnchor.constraint(equalTo: inviteOthersButton.superview!.trailingAnchor, constant: -standardOffset)
-        ]
+        if let superview = inviteOthersButton.superview {
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController+Constraints.swift]: When is this condition supposed to be true and when false?
If it's always supposed to evaluate to true we might not want to fail silently.

File: wire-ios/Wire-iOS/Sources/UserInterface/Helpers/KeyboardHeight.swift [hunk 2]
```diff
@@ -32,12 +32,19 @@ final class KeyboardHeight: NSObject {
     }
 
     private static func phoneKeyboardHeight() -> CGFloat {
+        let bottomInset = getSafeAreaBottomInset() ?? 0
         switch UIScreen.main.bounds.height {
         case 667: return 258
         case 736: return 271
-        case 812: return 253 + UIScreen.safeArea.bottom
+        case 812: return 253 + bottomInset
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Helpers/KeyboardHeight.swift:39]: We should get rid of this code eventually.

**Review by caldrian — APPROVED** [review]
The changes look good superficially. Please try to run the app on all possible devices.

File: wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController+Constraints.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController+Constraints.swift]: True when inviteOthersButton is already in the view hierarchy and has a parent view.
False when inviteOthersButton is not yet added to any parent view, and thus the superview is nil.

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ContactsUI/ContactsViewController+Constraints.swift]: Done: https://github.com/wireapp/wire-ios/commit/33dda8e7939078d0aced079c80c6beaeeaa62189

File: wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png
Comment (agisilaos) [reply wire-ios/Wire-iOS Tests/ReferenceImages/CameraKeyboardViewControllerTests/testCameraScrolledHorizontallyAwayPercent.1.png:1]: Already discussed this 1:1 :)

File: wire-ios/Wire-iOS/Sources/UserInterface/Helpers/KeyboardHeight.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Helpers/KeyboardHeight.swift:39]: Agreed!

**Pull request merged**